crates/astria-core/src/execution/v1alpha2/mod.rs

```diff
@@ -2,9 +2,107 @@ use prost_types::Timestamp;
 
 use crate::{
     generated::execution::v1alpha2 as raw,
+    sequencer::v1alpha1::{
+        IncorrectRollupIdLength,
+        RollupId,
+    },
     Protobuf,
 };
 
+// An error when transforming a [`raw::Block`] into a [`Block`].
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct GenesisInfoError(GenesisInfoErrorKind);
+
+impl GenesisInfoError {
+    fn incorrect_rollup_id_length(inner: IncorrectRollupIdLength) -> Self {
+        Self(GenesisInfoErrorKind::IncorrectRollupIdLength(inner))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum GenesisInfoErrorKind {
+    #[error("`rollup_id` field did not contain a valid rollup ID")]
+    IncorrectRollupIdLength(IncorrectRollupIdLength),
+}
+
+/// An Astria execution block on a rollup.
+///
+/// Contains information about the block number, its hash,
+/// its parent block's hash, and timestamp.
+///
+/// Usually constructed its [`Protobuf`] implementation from a
+/// [`raw::Block`].
+#[derive(Clone, Debug)]
+pub struct GenesisInfo {
+    /// The block number
+    rollup_id: RollupId,
+    /// The hash of the block
+    sequencer_genesis_block_number: u32,
```

Comment:
This should ideally be a `tendermint::block::Height`.

Reply:
I understand from our perspective, but that would require rollups import tendermint types, which we don't want to require.